### ISSUE TITLE
Use C++11 and Standalone ASIO as a compile time option

### ISSUE THREAD

**Issue by Type1J** [mologie/curl-asio#5]
I'd like to request an enhancement to the library be able to use C++11 and the standalone version of ASIO in place of Boost. With this option, it can be used in many more projects due to the widespread restriction on the usage of Boost.

I am not saying that it should use that option exclusively or even by default. This request is only an namespace alias request in the main library and perhaps the examples. For an example of what I mean, please see https://github.com/zaphoyd/websocketpp . Although it's a header only library, #define statements are used to use either Boost or C++11 headers, and similarly Boost.ASIO or ASIO.

**Comment by mologie** [comment]
I agree. I've too have dropped using Boost and used Asio in standalone mode ever since C++11 became widely supported by all platforms and compilers. While I have not used this library for quite some time now and there are some issues with some platforms and libcurl versions (see issues), I'll see to update it when I get the chance to.

**Comment by bdejong** [comment]
Fyi: We have meanwhile switched to using Casablanca from Microsoft. Cross
platform, modern and not dependent on curl - it's quite interesting. Also
has websockets support in the box.
